CRM/Contact/Form/Task/EmailTrait.php

```diff
@@ -660,4 +660,26 @@ public static function saveTemplateFormRule(array $fields) {
     return empty($errors) ? TRUE : $errors;
   }
 
+  /**
+   * Get selected contribution IDs.
+   *
+   * @return array
+   */
+  protected function getContributionIDs(): array {
+    return [];
+  }
+
+  /**
+   * Get case ID - if any.
+   *
+   * This raises questions -
+   * 1) can it be something other than an integer
+   * 2) is the right class for this the activity email task?
```

Comment:
For 1, it used to be - see https://lab.civicrm.org/dev/core/-/issues/2463

For 2, I don't have a thoughtful answer at the moment.

Reply:
@demeritcowboy so for clarity we could do

```
return $this->_caseId ? (int) $this->_caseId : NULL;
```

Reply:
@demeritcowboy for 2 - maybe I can copy the function onto that class & in this class have 

```
 protected function getCaseID() {
   if ($this->_caseId) {
    CRM_Core_Error::deprecatedWarning('case Id should only be present for case emails');
    return (int) $this->_caseId;
  } 
 return NULL
  }

```

Reply:
I've updated the PR to how I think it should look (as long as I'm right about case id being an activity only thing)

Reply:
I'm still thinking, but as a practical matter the deprecation comes up if you just send an email from Manage Case (i.e. from within the roles section), so it's obviously currently going thru there.

Reply:
Ah I thought that was the activity class - will track it down

Reply:
@demeritcowboy I toned it down to still keeping the case stuff on the Trait

Reply:
For the contribution one I get some warnings/notices, but it's pre-existing (e.g. find contributions - actions - send email). Looks good.